src/components/select/select.js

```diff
@@ -202,13 +203,19 @@ function SelectDirective($mdSelect, $mdUtil, $mdTheming, $mdAria, $compile, $par
       var valueEl = element.find('md-select-value');
       var isReadonly = angular.isDefined(attr.readonly);
 
+      mdSelectCtrl.mdSelectHeaderInput = element.find('md-select-header').find('input')[0];
```

Comment:
It's really unclear that this line is _only_ used in the check below. Perhaps move this until right before it's used (where the comment about ignoring the header inputs is).

Alternatively, can we change the selector that is used to find inputs to ignore the select-header? (I haven't looked how input is determined, but it seems like there should be an easier way.)